wallets/provider-ledger/src/helpers.ts

```diff
@@ -0,0 +1,54 @@
+import AppEth from '@ledgerhq/hw-app-eth';
+import TransportU2F from '@ledgerhq/hw-transport-webhid';
+
+const ETHEREUM_CHAIN_ID = '0x1';
+
+export const ETH_BIP32_PATH = "44'/60'/0'/0/0";
+
+const ledgerErrorMessages: { [statusCode: number | string]: string } = {
+  21781: 'The device is locked',
+  25871: 'Related application is not ready on your device',
+  27013: 'Action denied by user',
+  INSUFFICIENT_FUNDS: 'Insufficient funds for transaction',
+};
+
+export function getLedgerInstance() {
+  return { chainId: ETHEREUM_CHAIN_ID };
+}
+
+export async function getLedgerAccounts(): Promise<{
+  accounts: string[];
+  chainId: string;
+}> {
+  let transport = null;
+  try {
+    transport = await TransportU2F.create();
+
+    const eth = new AppEth(transport);
+
+    const accounts: string[] = [];
+
+    const result = await eth.getAddress(ETH_BIP32_PATH, false, true);
+    accounts.push(result.address);
+
+    return {
+      accounts: accounts,
+      chainId: ETHEREUM_CHAIN_ID,
+    };
+  } catch (error: any) {
+    throw new Error(getLedgerErrorMessage(error));
```

Comment:
I think you should change it to `throw getLedgerErrorMessage(error)`. 
And also change `getLedgerErrorMessage` function to something like this:
```js
if (errorStatusCode && !!ledgerErrorMessages[errorStatusCode]) {
    throw new Error(ledgerErrorMessages[errorStatusCode]);
  }
  throw error;
```

Reply:
Improved it. Thanks